### DISCUSSION TITLE
Hosting in a subdirectory

### DISCUSSION THREAD

**Discussion by sssionek** [outline/outline#5833] [Self Hosting]
Hi,
I set up Outline behind an NGINX reverse proxy inside a virtual subdirectory. The pages load just fine, but everything else still uses the base path (https://xxx/static/assets instead of https://xxx/outline/static/assets) and thus - not load at all. 
This is the NGINX location config:
```
location /outline {
                        proxy_pass http://localhost:3781;
                        proxy_buffering off;
                        proxy_set_header Upgrade $http_upgrade;
                        proxy_set_header Connection "Upgrade";
                        proxy_set_header Host $host;

                        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;proxy_set_header Host $host;
                        proxy_set_header Host $http_host;
                        proxy_set_header X-Real-IP $remote_addr;
                        proxy_set_header X-Scheme $scheme;
                        proxy_set_header X-Forwarded-Proto $scheme;
                        proxy_redirect off;
 }
```
Relevant entries from Outline's .env:
```
FORCE_HTTPS=false
URL=https://xxx/outline
PORT=3781
```
I was able to get the static assets to load by setting the CDN variable to some other directory and symlinking the build/assets folder there, but this only fixes a part of the issue, since /api and other endpoints still don't load. I don't really understand it, I thought setting the URL variable would be setting the base path.
Could someone nudge me in the right direction?

**Comment by tommoor** [comment]
I wouldn't attempt it, _lots_ of the frontend and backend code assumes that this is hosted at the root – there is no compatibility with hosting in a subdirectory I'm afraid.
